### PR TITLE
update dependency

### PR DESCRIPTION
**Pull request by esqi**
This PR update:
Kotlin Version to 1.6.21
AGP to 7.2.1
compile and target sdk to 32
build tools to 32.1.0-rc1
java to java 11
gradle to 7.4.2
protobuf lite to 1.47.0
androidx.compose.compiler:compiler to 1.1.1
androidx.activity:activity-compose to 1.4.0
add PendingIntent.FLAG_IMMUTABLE to PendingIntent
update room version to 2.4.2
update kotlin coroutine version to 1.6.2

### CODE REVIEW

File: config/library.gradle [hunk 1]
```diff
@@ -5,11 +5,11 @@ apply plugin: 'kotlin-android'
...
 
 android {
-    compileSdkVersion 30
-    buildToolsVersion "30.0.3"
+    compileSdkVersion 32
+    buildToolsVersion "32.1.0-rc1"
     defaultConfig {
-        minSdkVersion 21
-        targetSdkVersion 30
+        minSdkVersion 23
```
Comment (lukasdylan) [inline config/library.gradle:11]: ini kenapa naik?

Comment (esqi) [reply config/library.gradle:11]: butuh untuk PendingIntent.FLAG_IMMUTABLE

Comment (esqi) [reply config/library.gradle:11]: https://medium.com/androiddevelopers/all-about-pendingintents-748c8eb8619

Comment (lukasdylan) [reply config/library.gradle:11]: itu bukannya cuma tambahin `or PendingIntent.FLAG_IMMUTABLE` ya? cmiiw gw ga ada project buat ngetest grpcnya

Comment (lukasdylan) [reply config/library.gradle:11]: klo gw liat library [Chucker](https://github.com/ChuckerTeam/chucker/pull/593/files) sih begitu ya

Comment (esqi) [reply config/library.gradle:11]:
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
  PendingIntent.FLAG_IMMUTABLE
    } else {
        0
    }
```
pake itu paling

**Pull request closed without merging**